### PR TITLE
BUG: Make sure int is passed to pandas date index

### LINKED ISSUE #3349: BUG: pandas 0.19.2 compatibility - long versus integer in timestamp
This looks like a new failure with pandas 0.19.2 on appveyor windows python 2.7, it passes on python 3.5.
The unit tests passed with pandas 0.19.1 in previous run at https://ci.appveyor.com/project/josef-pkt/statsmodels/build/1.0.1076/job/863r6ag0kup7wspa

failing in a PR that is the latest test run:
https://ci.appveyor.com/project/josef-pkt/statsmodels/build/1.0.1094/job/7qac29xs07orl9pw

```

======================================================================
ERROR: statsmodels.tsa.statespace.tests.test_sarimax.test_misc_exog
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Py\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\tests\test_sarimax.py", line 1982, in test_misc_exog
    res.forecast(steps=1, exog=oos_exog)
  File "C:\projects\statsmodels\statsmodels\base\wrapper.py", line 95, in wrapper
    obj = data.wrap_output(func(results, *args, **kwargs), how)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2401, in forecast
    return self.predict(start=self.nobs, end=end, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2374, in predict
    prediction_results = self.get_prediction(start, end, dynamic, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\sarimax.py", line 1889, in get_prediction
    self.model._get_prediction_index(start, end, index, silent=True))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 447, in _get_prediction_index
    start, start_index, start_oos = self._get_index_label_loc(start)
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 383, in _get_index_label_loc
    self._get_index_loc(key, base_index))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 308, in _get_index_loc
    freq=base_index.freq)
  File "C:\Py\lib\site-packages\pandas\util\decorators.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 301, in __new__
    ambiguous=ambiguous)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 506, in _generate
    index = _generate_regular_range(start, end, periods, offset)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 1977, in _generate_regular_range
    dates = list(xdr)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2894, in generate_range
    end = start + (periods - 1) * offset
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 395, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 390, in __add__
    return self.apply(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 70, in wrapper
    result = func(self, other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2041, in apply
    other = other + relativedelta(months=3 * n - monthsSince, day=1)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 383, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 365, in __add__
    ret = (other.replace(**repl)
  File "pandas\tslib.pyx", line 697, in pandas.tslib.Timestamp.replace (pandas\tslib.c:14489)
  File "pandas\tslib.pyx", line 691, in pandas.tslib.Timestamp.replace.validate (pandas\tslib.c:14143)
ValueError: value must be an integer, received <type 'long'> for year
======================================================================
ERROR: statsmodels.tsa.statespace.tests.test_structural.test_misc_exog
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Py\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\tests\test_structural.py", line 334, in test_misc_exog
    res.forecast(steps=1, exog=oos_exog)
  File "C:\projects\statsmodels\statsmodels\base\wrapper.py", line 95, in wrapper
    obj = data.wrap_output(func(results, *args, **kwargs), how)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2401, in forecast
    return self.predict(start=self.nobs, end=end, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2374, in predict
    prediction_results = self.get_prediction(start, end, dynamic, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\structural.py", line 1486, in get_prediction
    self.model._get_prediction_index(start, end, index, silent=True))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 447, in _get_prediction_index
    start, start_index, start_oos = self._get_index_label_loc(start)
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 383, in _get_index_label_loc
    self._get_index_loc(key, base_index))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 308, in _get_index_loc
    freq=base_index.freq)
  File "C:\Py\lib\site-packages\pandas\util\decorators.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 301, in __new__
    ambiguous=ambiguous)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 506, in _generate
    index = _generate_regular_range(start, end, periods, offset)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 1977, in _generate_regular_range
    dates = list(xdr)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2894, in generate_range
    end = start + (periods - 1) * offset
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 395, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 390, in __add__
    return self.apply(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 70, in wrapper
    result = func(self, other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2041, in apply
    other = other + relativedelta(months=3 * n - monthsSince, day=1)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 383, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 365, in __add__
    ret = (other.replace(**repl)
  File "pandas\tslib.pyx", line 697, in pandas.tslib.Timestamp.replace (pandas\tslib.c:14489)
  File "pandas\tslib.pyx", line 691, in pandas.tslib.Timestamp.replace.validate (pandas\tslib.c:14143)
ValueError: value must be an integer, received <type 'long'> for year
======================================================================
ERROR: statsmodels.tsa.statespace.tests.test_varmax.TestVAR_exog.test_forecast
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Py\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\tests\test_varmax.py", line 437, in test_forecast
    desired = self.results.forecast(steps=16, exog=exog)
  File "C:\projects\statsmodels\statsmodels\base\wrapper.py", line 95, in wrapper
    obj = data.wrap_output(func(results, *args, **kwargs), how)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2401, in forecast
    return self.predict(start=self.nobs, end=end, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2374, in predict
    prediction_results = self.get_prediction(start, end, dynamic, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\varmax.py", line 781, in get_prediction
    self.model._get_prediction_index(start, end, index, silent=True))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 447, in _get_prediction_index
    start, start_index, start_oos = self._get_index_label_loc(start)
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 383, in _get_index_label_loc
    self._get_index_loc(key, base_index))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 308, in _get_index_loc
    freq=base_index.freq)
  File "C:\Py\lib\site-packages\pandas\util\decorators.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 301, in __new__
    ambiguous=ambiguous)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 506, in _generate
    index = _generate_regular_range(start, end, periods, offset)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 1977, in _generate_regular_range
    dates = list(xdr)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2894, in generate_range
    end = start + (periods - 1) * offset
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 395, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 390, in __add__
    return self.apply(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 70, in wrapper
    result = func(self, other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2041, in apply
    other = other + relativedelta(months=3 * n - monthsSince, day=1)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 383, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 365, in __add__
    ret = (other.replace(**repl)
  File "pandas\tslib.pyx", line 697, in pandas.tslib.Timestamp.replace (pandas\tslib.c:14489)
  File "pandas\tslib.pyx", line 691, in pandas.tslib.Timestamp.replace.validate (pandas\tslib.c:14143)
ValueError: value must be an integer, received <type 'long'> for year
======================================================================
ERROR: statsmodels.tsa.statespace.tests.test_varmax.TestVAR_exog2.test_forecast
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Py\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\tests\test_varmax.py", line 528, in test_forecast
    desired = self.results.forecast(steps=16, exog=exog)
  File "C:\projects\statsmodels\statsmodels\base\wrapper.py", line 95, in wrapper
    obj = data.wrap_output(func(results, *args, **kwargs), how)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2401, in forecast
    return self.predict(start=self.nobs, end=end, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2374, in predict
    prediction_results = self.get_prediction(start, end, dynamic, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\varmax.py", line 781, in get_prediction
    self.model._get_prediction_index(start, end, index, silent=True))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 447, in _get_prediction_index
    start, start_index, start_oos = self._get_index_label_loc(start)
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 383, in _get_index_label_loc
    self._get_index_loc(key, base_index))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 308, in _get_index_loc
    freq=base_index.freq)
  File "C:\Py\lib\site-packages\pandas\util\decorators.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 301, in __new__
    ambiguous=ambiguous)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 506, in _generate
    index = _generate_regular_range(start, end, periods, offset)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 1977, in _generate_regular_range
    dates = list(xdr)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2894, in generate_range
    end = start + (periods - 1) * offset
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 395, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 390, in __add__
    return self.apply(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 70, in wrapper
    result = func(self, other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2041, in apply
    other = other + relativedelta(months=3 * n - monthsSince, day=1)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 383, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 365, in __add__
    ret = (other.replace(**repl)
  File "pandas\tslib.pyx", line 697, in pandas.tslib.Timestamp.replace (pandas\tslib.c:14489)
  File "pandas\tslib.pyx", line 691, in pandas.tslib.Timestamp.replace.validate (pandas\tslib.c:14143)
ValueError: value must be an integer, received <type 'long'> for year
======================================================================
ERROR: statsmodels.tsa.statespace.tests.test_varmax.test_misc_exog
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Py\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\tests\test_varmax.py", line 817, in test_misc_exog
    res.forecast(steps=1, exog=oos_exog)
  File "C:\projects\statsmodels\statsmodels\base\wrapper.py", line 95, in wrapper
    obj = data.wrap_output(func(results, *args, **kwargs), how)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2401, in forecast
    return self.predict(start=self.nobs, end=end, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\mlemodel.py", line 2374, in predict
    prediction_results = self.get_prediction(start, end, dynamic, **kwargs)
  File "C:\projects\statsmodels\statsmodels\tsa\statespace\varmax.py", line 781, in get_prediction
    self.model._get_prediction_index(start, end, index, silent=True))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 447, in _get_prediction_index
    start, start_index, start_oos = self._get_index_label_loc(start)
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 383, in _get_index_label_loc
    self._get_index_loc(key, base_index))
  File "C:\projects\statsmodels\statsmodels\tsa\base\tsa_model.py", line 308, in _get_index_loc
    freq=base_index.freq)
  File "C:\Py\lib\site-packages\pandas\util\decorators.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 301, in __new__
    ambiguous=ambiguous)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 506, in _generate
    index = _generate_regular_range(start, end, periods, offset)
  File "C:\Py\lib\site-packages\pandas\tseries\index.py", line 1977, in _generate_regular_range
    dates = list(xdr)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2894, in generate_range
    end = start + (periods - 1) * offset
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 395, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 390, in __add__
    return self.apply(other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 70, in wrapper
    result = func(self, other)
  File "C:\Py\lib\site-packages\pandas\tseries\offsets.py", line 2041, in apply
    other = other + relativedelta(months=3 * n - monthsSince, day=1)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 383, in __radd__
    return self.__add__(other)
  File "C:\Py\lib\site-packages\dateutil\relativedelta.py", line 365, in __add__
    ret = (other.replace(**repl)
  File "pandas\tslib.pyx", line 697, in pandas.tslib.Timestamp.replace (pandas\tslib.c:14489)
  File "pandas\tslib.pyx", line 691, in pandas.tslib.Timestamp.replace.validate (pandas\tslib.c:14143)
ValueError: value must be an integer, received <type 'long'> for year

```

### PR DESCRIPTION
**Pull request by ChadFulton**
Closes #3349 

Used the fix @jreback suggested (thanks!).

### CODE REVIEW

File: statsmodels/tsa/base/tsa_model.py [hunk 1]
```diff
@@ -304,7 +304,8 @@ def _get_index_loc(self, key, base_index=None):
                 # Out-of-sample (note that we include key itself in the new
                 # index)
                 elif key > len(base_index) - 1:
-                    index = index_class(start=base_index[0], periods=key + 1,
```
Comment (jreback) [inline statsmodels/tsa/base/tsa_model.py]: only this one matters

**Comment by stonebig** [comment]
this patch is important to me. Will it be merge soon ?

**Comment by josef-pkt** [comment]
I thought Chad already merged it.

However, the appveyor test run on python 2.7 stopped with an error. So, maybe there is still a problem.
I see if I can restart the appveyor job.

**Comment by josef-pkt** [comment]
python 2.7 on appveyor is green after restarting, I will wait for python 3.5, then merge.

**Comment by josef-pkt** [comment]
no python 3.5 testrun stopped `Command exited with code -1073741819`
merging anyway because it was green on the previous run

**Comment by ChadFulton** [comment]
@josef-pkt Thanks for taking care of this, I forgot that I hadn't merged.

**Pull request merged**